tests/test_data_frame.py

```diff
@@ -584,5 +584,45 @@ def test_memory_usage(self):
                          base_df.memory_usage(deep=True, index=True).sum())
 
 
+    def test_attr_access(self):
+        base_df = pd.DataFrame([{'n': i, 's': (i*2)} for i in range(10)])
+        df = ps.DataFrame(base_df)
+
+        assertDataFrameEqualsPandas(df.n, pd.DataFrame(base_df.n))
+
+    def test_complex_write_attr_access(self):
+        base_df_1 = pd.DataFrame([
+            {'a': i, 'b': j, 'c': 100*i, 'd': -j}
+            for i in range(3) for j in range(3)
+        ])
+        base_df_2 = pd.DataFrame([
+            {'a': i, 'b': j, 'e': 50*i, 'f': j}
+            for i in range(3) for j in range(3)
+        ])
+        df_1 = ps.DataFrame(base_df_1)
+        df_2 = ps.DataFrame(base_df_2)
+
+        base_merged = base_df_1.merge(base_df_2, on=['a', 'b'])
+        base_merged['diff'] = base_merged['c'] - base_merged['e']
+        base_merged['key'] = base_merged['diff'] * \
+            (base_merged['d'] - base_merged['f'])
+        base_agg = base_merged.groupby('key', as_index=False)[['a', 'b']].sum()
+        base_agg['sum'] = base_agg['a'] + base_agg['b']
+        base_ordered = base_agg.sort_values(by='sum')
+
+        merged = df_1.merge(df_2, on=['a', 'b'])
+        merged['diff'] = merged['c'] - merged['e']
+        merged['key'] = merged['diff'] * \
+            (merged['d'] - merged['f'])
+        agg = merged.groupby('key')[['a', 'b']].sum()
+        agg['sum'] = agg['a'] + agg['b']
```

Comment:
I think it makes sense to use the attr-based access for all of these column-accesses too.